test/src/server/routes/collection.js

```diff
@@ -0,0 +1,221 @@
+			description: 'some description',
+			entityType: 'Author',
+			name: 'collectionName',
+			privacy: 'public'
+		};
+		const res = await agent.post('/collection/create/handler').send(data);
+		const collection = await new UserCollection({id: res.body.id}).fetch();
+
+		const res2 = await agent.get(`/search/search?q=${data.name}&type=collection`);
+		// TODO
+		expect(1).to.equal(1);
+	});
+
+	it('should return status code 500 for incorrect entityType', async () => {
+		const data = {
+			description: 'some description',
+			entityType: 'incorrect',
+			name: 'collectionName',
+			privacy: 'public'
+		};
+		const res = await agent.post('/collection/create/handler').send(data);
+		expect(res.status).to.equal(500);
+	});
+
+	it('should return status code 500 for empty collection name', async () => {
+		const data = {
+			description: 'some description',
+			entityType: 'Author',
+			name: '',
+			privacy: 'public'
+		};
+		const res = await agent.post('/collection/create/handler').send(data);
+		expect(res.status).to.equal(500);
+	});
+
+	it('should correctly add collaborators in the collection', async () => {
+		const collaborator1 = await createEditor();
+		const collaborator2 = await createEditor();
+		const data = {
+			collaborators: [collaborator1.toJSON(), collaborator2.toJSON()],
+			description: 'some description',
+			entityType: 'Author',
+			name: 'collectionName',
+			privacy: 'public'
+		};
+		const res = await agent.post('/collection/create/handler').send(data);
+		const collection = await new UserCollection({id: res.body.id}).fetch({withRelated: ['collaborators']});
+		const collectionJSON = collection.toJSON();
+		const collaboratorIds = collectionJSON.collaborators.map(collaborator => collaborator.collaboratorId);
+		expect(collaboratorIds).to.be.containingAllOf([collaborator1.get('id'), collaborator2.get('id')]);
+	});
+});
+
+describe('POST collection/edit', () => {
+	// eslint-disable-next-line one-var
+	let agent, collaborator, collectionJSON;
+	beforeEach(async () => {
+		try {
+			await createEditor(123456);
+		}
+		catch (error) {
+			// console.log(error);
+		}
+		// The `agent` now has the sessionid cookie saved, and will send it
+		// back to the server in the next request:
+		agent = await chai.request.agent(app);
+		await agent.get('/cb');
+		collaborator = await createEditor();
+		const data = {
+			collaborators: [collaborator.toJSON()],
+			description: 'some description',
+			entityType: 'Author',
+			name: 'collectionName',
+			privacy: 'public'
+		};
+		const res = await agent.post('/collection/create/handler').send(data);
+		const collection = await new UserCollection({id: res.body.id}).fetch({withRelated: ['collaborators']});
+		collectionJSON = collection.toJSON();
+	});
+	afterEach((done) => {
+		// Clear DB tables then close superagent server
+		truncateEntities().then(() => {
+			agent.close(done);
+		});
+	});
+
+	it('should update the collection and return 200 status code', async () => {
```

Comment:
I'd like to see this test split into three separate tests:

- one where you test changing the name, description, privacy, and check that each field has been changed properly like you did
- one where you test adding collaborators
- one where you test removing collaborators